File: MeatGrinder/Helpers/CustomAuthorize.cs
using System.Web;
using System.Web.Mvc;
using MeatGrinder.Services;

namespace MeatGrinder.Helpers
{
    public class CustomAuthorize : AuthorizeAttribute
    {
        protected override bool AuthorizeCore(HttpContextBase httpContext)
        {
            int userID = 0;

            var cookieValue = CookieService.GetCookie(HttpContext.Current, "UserID");
            if (cookieValue != null)
                userID = int.Parse(cookieValue);

            return userID != 0;
        }
    }
}

File: MeatGrinder.Web/Controllers/CustomAuthorize.fs
module MeatGrinder.Web.Helpers
open System
open MeatGrinder.Web.Services
type CustomAuthorize () =
    inherit System.Web.Mvc.AuthorizeAttribute()
    override x.AuthorizeCore(httpContext) =
        let userId = 0
        let cookieValue = CookieService.GetCookie httpContext "UserID"
        if cookieValue.IsSome && cookieValue.Value<>null then Int32.Parse(cookieValue.Value)<>0 else false


